### PR TITLE
feat: support Keep-Alive Header

### PR DESCRIPTION
**Pull request by fengmk2**
https://github.com/node-modules/agentkeepalive/issues/58

##### Checklist

- [ ] `npm test` passes
- [ ] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [ ] commit message follows commit guidelines

### CODE REVIEW

**Comment by fengmk2** [comment]
详细介绍文档 https://yuque.com/egg/nodejs/keep-alive-agent-econnreset

File: app/middleware/meta.js [hunk 1]
```diff
@@ -4,13 +4,24 @@
 
 'use strict';
 
-module.exports = options => {
+module.exports = (options, app) => {
+  let server;
+  app.once('server', s => {
```
Comment (atian25) [inline app/middleware/meta.js]: app 上没有直接获取 server 的?

File: app/middleware/meta.js [hunk 2]
```diff
@@ -4,13 +4,24 @@
...
       ctx.coreLogger.info('[meta] request started, host: %s, user-agent: %s', ctx.host, ctx.header['user-agent']);
     }
     await next();
     // total response time header
     ctx.set('x-readtime', Date.now() - ctx.starttime);
+
+    // try to support Keep-Alive Header
+    if (server && server.keepAliveTimeout && server.keepAliveTimeout >= 1000 && ctx.header.connection !== 'close') {
+      const timeout = parseInt(server.keepAliveTimeout / 1000);
+      ctx.set('keep-alive', `timeout=${timeout}`);
```
Comment (atian25) [inline app/middleware/meta.js]: 如果用户自己设置了这个 header 呢?

File: app/middleware/meta.js (hunk 1 above)
Comment (fengmk2) [reply app/middleware/meta.js]: 我直接加个 app.server 好了

File: app/middleware/meta.js (hunk 2 above)
Comment (fengmk2) [reply app/middleware/meta.js]: 这个 header 不允许应用设置的，必须保证它的正确性

**Comment by fengmk2** [comment]
增加 app.server

**Comment by popomore** [comment]
mm.app 是不是也考虑 emit server

**Comment by fengmk2** [comment]
@popomore 嗯，我加上

**Pull request merged**